pubsub/topics/main.go

```diff
@@ -145,6 +154,46 @@ func publish(client *pubsub.Client, topic, msg string) error {
 	return nil
 }
 
+func publishThatScales(client *pubsub.Client, topic string, n uint64) error {
+	ctx := context.Background()
+	// [START pubsub_publish_with_error_handling_that_scales]
+	var wg sync.WaitGroup
+	var totalErrors uint64
+	t := client.Topic(topic)
+
+	for i := 0; i < int(n); i++ {
+		result := t.Publish(ctx, &pubsub.Message{
+			// data must be a ByteString
+			Data: []byte("Message " + strconv.Itoa(i)),
+		})
+
+		wg.Add(1)
+		go func(i int, res *pubsub.PublishResult) {
+			defer wg.Done()
+			// The Get method blocks until a server-generated ID or
+			// an error is returned for the published message.
+			id, err := res.Get(ctx)
+			if err != nil {
+				// Error handling code can be added here.
+				log.Output(1, fmt.Sprintf("Failed to publish: %v", err))
+				atomic.AddUint64(&totalErrors, 1)
```

Comment:
Add a return here then remove the else? That way, only the error flow is indented. See https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow.

Reply:
Booked marked the Code Review link. Thanks a lot!